src/MongoDB/ReadPreference.c

```diff
@@ -264,6 +312,38 @@ static PHP_METHOD(ReadPreference, __set_state)
 	php_phongo_readpreference_init_from_hash(intern, props);
 } /* }}} */
 
+/* {{{ proto array|null MongoDB\Driver\ReadPreference::getHedge()
+   Returns the ReadPreference hedge document */
+static PHP_METHOD(ReadPreference, getHedge)
+{
+	php_phongo_readpreference_t* intern;
+	const bson_t*                hedge;
+
+	intern = Z_READPREFERENCE_OBJ_P(getThis());
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+
+	hedge = mongoc_read_prefs_get_hedge(intern->read_preference);
+
+	if (!bson_empty0(hedge)) {
+		php_phongo_bson_state state;
+
+		PHONGO_BSON_INIT_STATE(state);
+		state.map.root_type = PHONGO_TYPEMAP_NATIVE_ARRAY;
```

Comment:
Not sure if there's prior art for this: we want to return a hedge document. Do we use:
* a PHP `array`, which is the current solution:
```
array(1) {
  ["enabled"]=>
  bool(true)
}
```
* an object since it's a document?
```
array(stdClass)#%d (%d) {
  ["enabled"]=>
  bool(true)
}
```

Type map support is definitely not warranted here, and with arrays being what the average PHP developer would use here I would keep the current array return. PHONGO_BSON_INIT_DEBUG_STATE would use that for arrays and documents as well, in the off-chance that the hedge document would get more complex in the future.

Reply:
I think returning an object is fine. On the off chance someone actually needs to use this method and work with its return value, it's not too difficult to cast to an array or apply their own type map. Prior art would be `Javascript::getScope()` and the `getInfo()` methods for WriteConcernError and WriteError. I believe all of those code paths use `PHONGO_BSON_INIT_STATE` via `php_phongo_bson_to_zval`. 

`Server::getInfo()` may be an exception for returning an array, but we've historically never documented that it is a copy of the most recent `isMaster` document. It's always been thought of as a debug method.

Reply:
Changed to return objects in all cases.